src/components/PostDetails.tsx

```diff
@@ -1,100 +1,53 @@
 import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as commentsAction from '../features/comments';
 
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+export const PostDetails: React.FC = () => {
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
+  const dispatch = useAppDispatch();
 
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
+  const {
+    comments,
+    loaded,
+    error: hasError,
+  } = useAppSelector(state => state.comments);
 
-  useEffect(loadComments, [post.id]);
+  const { selectedPost: post } = useAppSelector(state => state.selectedPost);
```

Comment:
Verify that the `state.selectedPost` structure matches the actual Redux state structure. If the state does not have a `selectedPost` property, this will cause an error.